test/test_cuda.py

```diff
@@ -1678,36 +1677,76 @@ def test_events(self):
         self.assertGreater(start_event.elapsed_time(event), 0)
 
     @staticmethod
-    def _stream_synchronize(spin_time=50000000):
+    def _stream_synchronize(self, spin_time=50000000):
```

Comment:
nit:  can you rename spin_time to have units? `spin_time_us` or whatever it should be (I don't know)